functions.php

```diff
@@ -28,14 +38,59 @@ function DBCheck($data): string {
 
 function makePlantEntryTile(array $entry): string {
     if (empty($entry)) {
-        return 'There is no data for this entry';
+        $return_string = 'There is no data for this entry';
     } elseif (count($entry) < 4) {
-        return 'There is not enough information given for this type of entry.';
+        $return_string = 'There is not enough information given for this type of entry.';
     } elseif (!empty($entry)) {
-        return "<div class='entry_box'>
+        $return_string = "<div class='entry_box'>
                         <div class='entry science_name'>" . $entry['science_name'] . "</div>
-                        <div class='entry'>" . $entry['name'] . "</div>
-                        <div class='entry'>" . $entry['type'] . "</div>
+                        <div class='entry'>" . ucwords($entry['name'],"/ -") . "</div>
+                        <div class='entry'>" . ucwords(strtolower($entry['type']),"/ -") . "</div>
                     </div>";
     }
+
+    return $return_string;
+}
+
+function listPlantTypes(array $plant_types): string {
+    if (array_key_exists('type',$plant_types[0])) {
+        $type_echo = '<ul>';
+        foreach ($plant_types as $entry) {
+            $type_echo.= '<li>' . ucwords(strtolower($entry['type']),"/ -") . '</li>';
+        }
+        $type_echo.= '</ul>';
+    } else {
+        $type_echo = 'Array key error. Please enter different data.';
+    }
+
+    return $type_echo;
+}
+
+function checkforUniqueAddEntry($db, string $science_name): bool {
+    $nonunique_entry_query = $db->prepare("SELECT `id` FROM `plants` WHERE `science_name` = ?;");
+    $nonunique_entry_query->execute([$science_name]);
+    $nonunique_entry = $nonunique_entry_query->fetch();
+    if ($nonunique_entry !== false) {
+        $nonunique_entry = true;
+    }
+
+    return $nonunique_entry;
 }
+
+function checkForWrongPlantType($db, string $type): array {
+    $used_type_query = $db->prepare("SELECT `id` FROM `plant_types` WHERE `type` = upper(?);");
+    $used_type_query->execute([$type]);
+    $used_type = $used_type_query->fetch();
+    if ($used_type === false) {
+        $used_type = ['not found'];
+    }
+
+    return $used_type;
+}
+
+function insertDataToDB($db, string $science_name, string $name, array $used_type): string {
+    $add_entry_query = $db->prepare("INSERT INTO `plants` (`science_name`,`name`,`type`) VALUES (?,?,?);");
```

Comment:
please use named placeholders for this. if your names match the array keys you can just pass the array straight into execute